PyInquirer/prompts/list.py

```diff
@@ -67,118 +63,121 @@ def _init_choices(self, choices, default=None):
 
         # prepare the select choices
         for i, choice in enumerate(self.choices):
             append(i, choice)
         tokens.pop()  # Remove last newline.
         return tokens
 
+    def is_selection_a_separator(self):
+        selected = self.choices[self.selected_option_index]
+        return isinstance(selected[0], Separator)
+
+    def is_selection_disabled(self):
+        return self.choices[self.selected_option_index][2]
+
+    def is_selection_valid(self):
+        return (not self.is_selection_disabled() and
+                not self.is_selection_a_separator())
+
+    def select_next(self):
+        self.selected_option_index = (
+            (self.selected_option_index - 1) % self.choice_count)
+
+    def select_previous(self):
+        self.selected_option_index = (
+            (self.selected_option_index + 1) % self.choice_count)
+
     def get_selection(self):
         return self.choices[self.selected_option_index]
 
 
-def question(message, **kwargs):
-    # TODO disabled, dict choices
-    if not 'choices' in kwargs:
-        raise PromptParameterException('choices')
-
-    choices = kwargs.pop('choices', None)
-    default = kwargs.pop('default', 0)  # TODO
-    qmark = kwargs.pop('qmark', '?')
-    # TODO style defaults on detail level
-    style = kwargs.pop('style', default_style)
+def question(message,
+             choices,
+             default=0,
+             qmark="?",
+             style=None,
+             **kwargs):
+    merged_style = merge_styles([DEFAULT_STYLE, style])
 
     ic = InquirerControl(choices)
 
-    def get_prompt_tokens(cli):
+    def get_prompt_tokens():
         tokens = []
 
-        tokens.append((Token.QuestionMark, qmark))
-        tokens.append((Token.Question, ' %s ' % message))
+        tokens.append(("class:qmark", qmark))
+        tokens.append(("class:question", ' {} '.format(message)))
         if ic.answered:
-            tokens.append((Token.Answer, ' ' + ic.get_selection()[0]))
+            tokens.append(("class:answer", ' ' + ic.get_selection()[0]))
         else:
-            tokens.append((Token.Instruction, ' (Use arrow keys)'))
+            tokens.append(("class:instruction", ' (Use arrow keys)'))
+
         return tokens
 
-    # assemble layout
-    layout = HSplit([
-        Window(height=D.exact(1),
-               content=TokenListControl(get_prompt_tokens)
-        ),
+    ps = PromptSession(get_prompt_tokens, reserve_space_for_menu=0)
+
+    layout = Layout(HSplit([
+        ps.layout.container,
         ConditionalContainer(
             Window(ic),
             filter=~IsDone()
         )
-    ])
+    ]))
 
-    # key bindings
-    manager = KeyBindingManager.for_prompt()
+    bindings = KeyBindings()
 
-    @manager.registry.add_binding(Keys.ControlQ, eager=True)
-    @manager.registry.add_binding(Keys.ControlC, eager=True)
+    @bindings.add(Keys.ControlQ, eager=True)
+    @bindings.add(Keys.ControlC, eager=True)
     def _(event):
```

Comment:
There's a bug here. It throwing a KeyboardInterrupt() event which is being cacth and this makes the current loop hard to exit. You have to answer all the questions.

Reply:
I've fixed the keyboard interrupt - it should now be handled they way it was before.